src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -308,7 +318,7 @@ public Heap()
         /// <param name="values">Array elements whose indices are integer</param>
         /// <param name="dict">Array elements whose indices are not integer</param>
         /// <returns></returns>
-        public StackValue AllocateArray(IEnumerable<StackValue> values)
+        public StackValue AllocateArray(StackValue[] values)
```

Comment:
Right, so you're trading the cost of having to allocate/copy new data (i.e. `ToArray` cost) with double-evaluation? Not too sure if that justifies it though ;)